File: php/src/MBPay/Exception/MBPayException.php
<?php

namespace MBPay\Exception;

use Exception;

/**
 * MBPay API 自定义异常
 */
class MBPayException extends Exception
{
    /**
     * @var int 错误码
     */
    private $errorCode;

    /**
     * @param int $code 错误码
     * @param string $message 错误信息
     */
    public function __construct(int $code, string $message)
    {
        $this->errorCode = $code;
        parent::__construct("MBPay API Error [{$code}]: {$message}", $code);
    }

    /**
     * 获取错误码
     *
     * @return int
     */
    public function getErrorCode(): int
    {
        return $this->errorCode;
    }
}

File: java/src/main/java/com/mbpay/exception/MBPayException.java
package com.mbpay.exception;

/**
 * MBPay API 自定义异常
 */
public class MBPayException extends Exception {
    private int errorCode;

    public MBPayException(int code, String message) {
        super("MBPay API Error [" + code + "]: " + message);
        this.errorCode = code;
    }

    /**
     * 获取错误码
     *
     * @return 错误码
     */
    public int getErrorCode() {
        return errorCode;
    }
}






